### PR TITLE
Create CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by JackTemaki**
This is text that I took from the i6 Wiki about programming guidelines.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+Coding conventions
+
+- use clangformat
+- no tabs, 4 spaces per indentation level
+- camel case for files and directories (start with a capital)
+- camel case for variables (classes: start with capital, everything else: start with lower case)
+- avoid abbreviations
+- avoid subdirectories
+- private variables should end with an underscore
```
Comment (curufinwe) [inline CONTRIBUTING.md:12]: private members (not private variables)

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+// The rest of the file
+
+#endif
+```
+
+Source Code Layout
+------------------
+
+**Rule**: 4 space indentation, no tabs
```
Comment (curufinwe) [inline CONTRIBUTING.md:40]: duplicate from above

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+- All code lines must be properly indented according to their block nesting.
+---
+**Rule**: Special characters like page break must not be used.
+ - Rationale: These characters are bound to cause problem for editors, printers, terminal emulators or debuggers when used in a multi-programmer, multi-platform environment. 
+---
+**Recommendation**: Logical units within a block should be separated by one blank line. 
+
+---
+**Rule**: Whenever a block exceeds one screen-full in length the closing brace should be commented.
```
Comment (curufinwe) [inline CONTRIBUTING.md:49]: how much is a screen?

Comment (JackTemaki) [reply CONTRIBUTING.md:49]: If you do not know no one knows I guess... For my screen it is about 70 so I will put that.

Comment (curufinwe) [reply CONTRIBUTING.md:49]: :ok:

**Review by curufinwe — APPROVED** [review]

**Review by SimBe195 — APPROVED** [review]

**Pull request merged**